libraries/abstractions/common_io/include/iot_uart.h

```diff
@@ -108,74 +109,102 @@ typedef enum
-    uint32_t ulWordlength;  /**< The word length to be set for the UART port. */
-    uint32_t ulStopbits;    /**< The stop bits to be set for the UART port. */
-    uint32_t ulFlowControl; /**< The flow control to be set for the UART port. */
+    uint32_t ulBaudrate;   /**< The baud rate to be set for the UART port. */
+    uint8_t ucParity;      /**< The parity to be set for the UART port: 0 is no parity, 1 is even parity and 2 is odd parity. */
+    uint8_t ucWordlength;  /**< The word length to be set for the UART port. */
+    uint8_t ucStopbits;    /**< The stop bits to be set for the UART port: 0 is no stop bits and 1 is stop bits. */
+    uint8_t ucFlowControl; /**< The flow control to be set for the UART port: 0 is disabled and 1 is enabled. */
 } IotUARTConfig_t;
 
 /**
- * @brief Initializes the UART peripheral of the board with default configuration.
+ * @brief Initializes the UART peripheral of the board.
  *
  * The application should call this function to initialize the desired UART port.
  *
+ * @warning Once opened, the same UART instance must be closed before calling open again.
+ *
  * @param[in] lUartInstance The instance of the UART port to initialize.
  *
- * @return handle to the UART if everything succeeds, otherwise NULL.
+ * @return
+ * - 'the handle to the UART port (not NULL)', on success.
+ * - 'NULL', if
+ *     - invalid instance number
+ *     - open same instance more than once before closing it
  */
 IotUARTHandle_t iot_uart_open( int32_t lUartInstance );
 
 /**
  * @brief Sets the application callback to be called on completion of an operation.
  *
- * On completion of receive or transmit on UART, the application is notified with a
- * function callback with the state of the transaction. The callback function and the parameters
- * for the state of the callback are set using iot_uart_set_callback.
- * The application should call this function with the callback function and its parameters.
+ * The callback is guaranteed to be invoked when the current asynchronous operation completes, either successful or failed.
+ * This simply provides a notification mechanism to user's application. It has no impact if the callback is not set.
+ *
+ * @note This callback will not be invoked when synchronous operation completes.
+ * @note This callback is per handle. Each instance has its own callback.
+ * @note Single callback is used for both read_async and write_async. Newly set callback overrides the one previously set.
+ * @warning If the input handle is invalid, this function silently takes no action.
  *
  * @param[in] pxUartPeripheral The peripheral handle returned in the open() call.
- * @param[in] xCallback The callback function to be called on completion of transaction.
- * @param[in] pvUserContext The user context to be passed back when callback is called.
+ * @param[in] xCallback The callback function to be called on completion of transaction (This can be NULL).
```

Comment:
Why do we have to say it can be NULL? If its NULL, then the function is a no-op

Reply:
IMO in case an implementation assumes that the callback could never be NULL and a user tries to change it to NULL for a no-op. I think it is a valid case that an application wants to change the callback to NULL without closing the handle. What are your thoughts?

Reply:
Yes, understood that it can be NULL for no-op or not set the callback are the same. But do we need to specify that "This can be NULL"?

Reply:
I think we should or add a test in order to enforce the behavior